commands/summary.go

```diff
@@ -58,224 +58,255 @@ func (res *summaryResult) String() string {
+			continue
+		}
+
+		addData(zipW, path.Join("server", rf.Name), rf.Modified, rc)
+
+		rc.Close() //nolint:errcheck
 	}
+}
 
-	b, err := json.MarshalIndent(status, "", "  ")
+// getURL returns `GET url` response body.
+func getURL(ctx context.Context, url string) ([]byte, error) {
+	req, err := http.NewRequestWithContext(ctx, "GET", url, nil)
 	if err != nil {
-		logrus.Debugf("%s", err)
-		b = []byte(err.Error())
-	}
-	b = append(b, '\n')
-	w, err := zipW.CreateHeader(&zip.FileHeader{
-		Name:     "client/status.json",
-		Method:   zip.Deflate,
-		Modified: time.Now(),
-	})
-	if err == nil {
-		_, err = w.Write(b)
+		return nil, errors.WithStack(err)
 	}
+	resp, err := http.DefaultClient.Do(req)
 	if err != nil {
-		logrus.Debugf("%s", err)
+		return nil, errors.WithStack(err)
 	}
+	defer resp.Body.Close() //nolint:errcheck
 
-	w, err = zipW.CreateHeader(&zip.FileHeader{
-		Name:     "client/pmm-admin-version.txt",
-		Method:   zip.Deflate,
-		Modified: time.Now(),
-	})
-	if err == nil {
-		_, err = w.Write([]byte(version.FullInfo()))
+	if resp.StatusCode != http.StatusOK {
+		return nil, errors.Errorf("status code: %d", resp.StatusCode)
 	}
+
+	b, err := ioutil.ReadAll(resp.Body)
 	if err != nil {
-		logrus.Debugf("%s", err)
+		return nil, errors.Wrap(err, "cannot read response body")
 	}
+	return b, nil
+}
 
-	// FIXME get it via pmm-agent's API - it is _not_ a good idea to use exec there
-	// golangli-lint should continue complain about it until it is fixed
-	b, err = exec.Command("pmm-agent", "--version").CombinedOutput()
-	if err != nil {
-		logrus.Debugf("%s", err)
-		b = []byte(err.Error())
+type pprofData struct {
+	name string
+	data []byte
+}
+
+// addPprofData adds pprof data to zip file.
+func addPprofData(ctx context.Context, zipW *zip.Writer, skipServer bool) {
+	profiles := []struct {
+		name    string
+		urlPath string
+	}{
+		{
+			"profile.pb.gz",
+			"/profile?seconds=60",
+		}, {
+			"heap.pb.gz",
+			"/heap?gc=1",
+		}, {
+			"trace.out",
+			"/trace?seconds=10",
+		},
 	}
-	w, err = zipW.CreateHeader(&zip.FileHeader{
-		Name:     "client/pmm-agent-version.txt",
-		Method:   zip.Deflate,
-		Modified: time.Now(),
-	})
-	if err == nil {
-		_, err = w.Write(b)
+
+	sources := map[string]string{
+		"client/pprof/pmm-agent": "http://127.0.0.1:7777/debug/pprof",
 	}
-	if err != nil {
-		logrus.Debugf("%s", err)
+	if !skipServer {
+		sources["server/pprof/pmm-managed"] = "http://127.0.0.1:7773/debug/pprof"
+		sources["server/pprof/qan-api2"] = "http://127.0.0.1:9933/debug/pprof"
 	}
 
-	addClientFile(zipW, status.ConfigFilepath)
+	for _, p := range profiles {
+		// fetch the same profile from different sources in parallel
+
+		var wg sync.WaitGroup
+		ch := make(chan pprofData, len(sources))
+
+		for dir, urlPrefix := range sources {
+			wg.Add(1)
+
+			go func(url, name string) {
+				defer wg.Done()
 
-	addClientCommand(zipW, "list.txt", &listCommand{NodeID: status.RunsOnNodeID})
+				logrus.Infof("Getting %s ...", url)
```

Comment:
It confuses that we write this in info level and don't write any info about finishing that or having an error in info level. Shouldn't we?